ocaml-lsp-server/src/custom_requests/req_infer_intf.mli

```diff
@@ -4,4 +4,8 @@ val capability : string * Json.t
 
 val meth : string
 
-val on_request : params:Jsonrpc.Structured.t option -> State.t -> Json.t Fiber.t
+val on_request :
+     params:Jsonrpc.Structured.t option
+  -> State.t Server.t
```

Comment:
why do we add this unused argument everywhere?

Reply:
It is necessary to be able to do some logging, similar to what is done in the hover's handler.

```
State.log_msg server ~type_:Warning ~message
```

And with the factorized version I need it to be able to call hover's handler from hoverExtend.

Also I tend to think that it is better if custom requests tend to mimic normal requests. Which means it's better if they receive `State.t Server.t` than just `State.t`. I tried to do a minimal chance. A more thorough and clean version would be

```diff
- val on_request : params:Jsonrpc.Structured.t option -> State.t -> Json.t Fiber.t
+ val on_request : params:Jsonrpc.Structured.t option -> State.t Server.t -> Json.t Fiber.t`
```

Reply:
If it's needed for logging, let's pass it to functions that actually log something. I see this argument being added in a few places without it actually being used.

> Also I tend to think that it is better if custom requests tend to mimic normal requests. Which means it's better if they receive State.t Server.t than just State.t. I tried to do a minimal chance. A more thorough and clean version would be

Some sort of consistency of type signature to me is a much lesser goal than making signatures as precise and as informative as possible.